src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.PkgDefFileReader.cs

```diff
@@ -0,0 +1,301 @@
+                                }
+
+                                return new PkgDefItem(sectionName, valueName,
+                                                        System.Text.Encoding.UTF8.GetString(binaryData),
+                                                        PkgDefItem.PkgDefValueType.MultiSz);
+                            }
+                        default:
+                            break;
+                    }
+
+                    return new PkgDefItem(sectionName, valueName, binaryData, PkgDefItem.PkgDefValueType.Binary);
+                }
+
+                if (valueDataString.StartsWith(Constants.DwordPrefix))
+                {
+                    var numericString = valueDataString.Substring(Constants.DwordPrefix.Length);
+                    if (numericString.Length > 8)
+                    {
+                        return null;
+                    }
+
+                    var dword = Convert.ToUInt32(numericString, 16);
+                    return new PkgDefItem(sectionName, valueName, dword, PkgDefItem.PkgDefValueType.DWord);
+                }
+
+                if (valueDataString.StartsWith(Constants.QwordPrefix))
+                {
+                    var numericString = valueDataString.Substring(Constants.QwordPrefix.Length);
+                    if (numericString.Length > 16)
+                    {
+                        return null;
+                    }
+
+                    var qword = Convert.ToUInt64(numericString, 16);
+                    return new PkgDefItem(sectionName, valueName, qword, PkgDefItem.PkgDefValueType.QWord);
+                }
+
+                return null;
+            }
+
+            private void SetNewSection(string line)
+            {
+                _currentSection = ParseSectionName(line);
+            }
+
+            private bool IsSectionLine(string line)
+            {
+                return line.StartsWith(Constants.SectionStartChar);
+            }
+
+            private bool StripQuotes(ref string line)
+            {
+                if ((line.Length > 1) && line.StartsWith("\"") && line.EndsWith("\""))
+                {
+                    line = line.Substring(1, line.Length - 2);
+                    return true;
+                }
+
+                return false;
+            }
+
+            private byte[] TransformHexToBinary(string hexString)
+            {
+                var normalizedString = string.Empty;
+
+                var curPos = 0;
+                while (curPos < hexString.Length)
+                {
+                    if (!IsValidHexChar(hexString[curPos]))
+                    {
+                        curPos++;
+                        continue;
+                    }
+
+                    if ((curPos + 1 >= hexString.Length) ||
+                        !IsValidHexChar(hexString[curPos + 1]))
+                    {
+                        //expecting a 2nd hex character
+                        //throw?
```

Comment:
if it's unrecoverable maybe throw, otherwise if you think it could cause other issues down the line it might be good to log telemetry?